cfn-resources/autogen/schema-gen/main.go

```diff
@@ -0,0 +1,587 @@
+				readOnly = append(readOnly, readOnlyDef...)
+				ids = append(ids, idsDef...)
+
+				//Required Props Query Params
+				requiredParams.UpdateFields.RequiredParams = requiredOnlyProperties(method.Parameters)
+				//Required Props from Body
+				requiredParams.UpdateFields.RequiredParams = append(requiredParams.UpdateFields.RequiredParams, capitalizeArray(reqParams)...)
+				//All Props from Body
+				requiredParams.UpdateFields.InputParams = inputOnlyProperties(bodySchema[key])
+
+			}
+
+			if method := b.Get; method != nil {
+				//Read from Req params
+				reqSchemaKey, reqSchema, reqDefinitions, reqParams := readRequestBody(method, doc)
+				//Read from Response params
+				resSchemaKey, resSchema, resDefinitions := readResponseBody(method, doc)
+
+				//Read from query params
+				queryParams := readQueryParams(method)
+				bodySchema[key] = mergePropertyMaps(bodySchema[key], mergePropertyMaps(reqSchema[reqSchemaKey], resSchema[resSchemaKey]))
+
+				//Merge all params
+				allMethodProps[key] = mergePropertyMaps(allMethodProps[key], mergePropertyMaps(bodySchema[key], queryParams))
+
+				definitions = mergeDefinitionMaps(definitions, mergeDefinitionMaps(reqDefinitions, resDefinitions))
+
+				definitions = defaultDefinition(definitions)
+				readOnly, ids = readOnlyAndUniqueProperties(bodySchema)
+				readOnlyDef, idsDef = readOnlyAndUniqueDefinitions(definitions)
+				readOnly = append(readOnly, readOnlyDef...)
+				ids = append(ids, idsDef...)
+
+				//Required Props Query Params
+				requiredParams.ReadFields.RequiredParams = requiredOnlyProperties(method.Parameters)
+				//Required Props from Body
+				requiredParams.ReadFields.RequiredParams = append(requiredParams.ReadFields.RequiredParams, capitalizeArray(reqParams)...)
+				//All Props from Body
+				requiredParams.ReadFields.InputParams = inputOnlyProperties(bodySchema[key])
+
+			}
+			if method := b.Delete; method != nil {
+				//Read from query params
+				for _, pRef := range method.Parameters {
+					newProps := readProperty(pRef.Value)
+					for k, property := range newProps {
+						if contains(k, optionalInputParams) {
+							continue
+						}
+						queryParams[capitalize(k)] = property
+					}
+				}
+				//Merge body params and query params
+				allMethodProps[key] = mergePropertyMaps(allMethodProps[key], queryParams)
+
+				//Required Props Query Params
+				requiredParams.DeleteFields.RequiredParams = requiredOnlyProperties(method.Parameters)
+
+			}
+		}
+		allMethodProps[key] = defaultProperty(allMethodProps[key])
+
+		requiredParams.FileName = res.TypeName
+		cfn = CfnSchema{
+			AdditionalProperties: false,
+			Definitions:          definitions,
+			Description:          description,
+			Handlers:             h,
+			PrimaryIdentifier:    ids,
+			Properties:           allMethodProps[key],
+			ReadOnlyProperties:   readOnly,
+			Required:             createReqParams,
+			TypeName:             MongoDBAtlasPrefix + typeName,
```

Comment:
the type name is generating in lowercase, mybe we can turn it into upper like this

```suggestion
						TypeName:             MongoDBAtlasPrefix + cases.Title(language.Und).String(typeName)
```